src/github.com/getlantern/fronted/direct.go

```diff
@@ -76,26 +77,28 @@ func (d *direct) vet(initial map[string][]*Masquerade) {
 			d.candidates <- arr[r]
 		}
 	}
+}
 
-	// Vet in parallel
-	for i := 0; i < VetParallelism; i++ {
-		go d.vetSome()
+func (d *direct) vetInitial() {
+	log.Debugf("Vetting %d initial candidates in parallel", NumberToVetInitially)
+	for i := 0; i < NumberToVetInitially; i++ {
+		go d.vetOne()
 	}
 }
 
-func (d *direct) vetSome() {
+func (d *direct) vetOne() {
```

Comment:
This goroutine now terminates once it has successfully verified one masquerade.  So, when we reconfigure, we don't end up piling up a bunch of goroutines.  Also, we don't do a bunch of work vetting masquerades that we may never use.